sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/interfaces.scala

```diff
@@ -119,18 +125,24 @@ case class AggregateExpression(
       normalizedAggFunc.canonicalized.asInstanceOf[AggregateFunction],
       mode,
       isDistinct,
+      filter,
```

Comment:
should we also get canonicalized filter?

Reply:
I don't quite understand the meaning of canonicalized filter.

Reply:
`filter.map(_. canonicalized)`, like what we do for the `normalizedAggFunc`

Reply:
OK. I will change it.